File: frontend/client/services/dtos/game-team-status.js

import GameLeader from './game-leader'
import { Num, Text, DTO } from 'services/util/types'

export default DTO({
  wins:           Num('1'),
  score:          Num('2'),
  losses:         Num('3'),
  teamId:         Text('4'),
  triCode:        Text('5'),
  name:           Text('6'),
  city:           Text('7'),
  splashUrl:      Text('8'),
  primaryColor:   Text('9'),
  secondaryColor: Text('10'),
  pointsLeader:   GameLeader('11'),
  assistsLeader:  GameLeader('12'),
  reboundsLeader: GameLeader('13'),
})

File: backend/dtos/game_team_status.go
package dtos

// GameTeamStatus is the status of a team in a game.
//
// easyjson:json
type GameTeamStatus struct {
	Wins                 int        `json:"1"`
	Score                int        `json:"2"`
	Losses               int        `json:"3"`
	TeamID               string     `json:"4"`
	TriCode              string     `json:"5"`
	Name                 string     `json:"6"`
	City                 string     `json:"7"`
	SplashURL            string     `json:"8"`
	SplashPrimaryColor   string     `json:"9"`
	SplashSecondaryColor string     `json:"10"`
	PointsLeader         GameLeader `json:"11"`
	AssistsLeader        GameLeader `json:"12"`
	ReboundsLeader       GameLeader `json:"13"`
}
